File: python-midonetclient/src/midonetclient/host_interface_port.py
# vim: tabstop=4 shiftwidth=4 softtabstop=4

# Copyright 2013 Midokura PTE LTD.
# All Rights Reserved
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may
# not use this file except in compliance with the License. You may obtain
# a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations
# under the License.


from midonetclient import resource_base
from midonetclient import vendor_media_type


class HostInterfacePort(resource_base.ResourceBase):

    media_type = vendor_media_type.APPLICATION_HOST_INTERFACE_PORT_JSON

    def __init__(self, uri, dto, auth):
        super(HostInterfacePort, self).__init__(uri, dto, auth)

    def get_host_id(self):
        return self.dto['hostId']

    def get_interface_name(self):
        return self.dto['interfaceName']

    def get_port_id(self):
        return self.dto['portId']

    def port_id(self, port_id):
        self.dto['portId'] = port_id
        return self

    def interface_name(self, name):
        self.dto['interfaceName'] = name
        return self

File: midonet-cluster/src/test/java/org/midonet/client/resource/HostInterfacePort.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.client.resource;

import java.net.URI;
import java.util.UUID;

import org.midonet.client.WebResource;
import org.midonet.client.dto.DtoHostInterfacePort;
import org.midonet.cluster.services.rest_api.MidonetMediaTypes;

public class HostInterfacePort
    extends ResourceBase<HostInterfacePort, DtoHostInterfacePort> {

    public HostInterfacePort(WebResource resource, URI uriForCreation,
                             DtoHostInterfacePort interfacePort) {
        super(resource, uriForCreation, interfacePort,
              MidonetMediaTypes.APPLICATION_HOST_INTERFACE_PORT_JSON());
    }

    @Override
    public URI getUri() {
        return principalDto.getUri();
    }

    public UUID getHostId() {
        return principalDto.getHostId();
    }

    public String getInterfaceName() {
        return principalDto.getInterfaceName();
    }

    public UUID getPortId() {
        return principalDto.getPortId();
    }

    public HostInterfacePort portId(UUID id) {
        principalDto.setPortId(id);
        return this;
    }

    public HostInterfacePort interfaceName(String name) {
        principalDto.setInterfaceName(name);
        return this;
    }
}
